axes/handlers/database.py

```diff
@@ -119,58 +119,55 @@ def user_login_failed(
-                client_str,
-                attempt.failures_since_start,
-                get_failure_limit(request, credentials),
-            )
-
             separator = "\n---------\n"
 
-            attempt.get_data = Concat("get_data", Value(separator + get_data))
-            attempt.post_data = Concat("post_data", Value(separator + post_data))
-            attempt.http_accept = request.axes_http_accept
-            attempt.path_info = request.axes_path_info
-            attempt.failures_since_start += 1
-            attempt.attempt_time = request.axes_attempt_time
-            attempt.save()
-        except AccessAttempt.DoesNotExist:
             # Record failed attempt with all the relevant information.
             # Filtering based on username, IP address and user agent handled elsewhere,
             # and this handler just records the available information for further use.
-
-            if not (settings.AXES_ONLY_USER_FAILURES and username is None):
+            if created:
                 log.warning(
                     "AXES: New login failure by %s. Creating new record in the database.",
                     client_str,
                 )
-                AccessAttempt.objects.create(
-                    username=username,
-                    ip_address=request.axes_ip_address,
-                    user_agent=request.axes_user_agent,
-                    get_data=get_data,
-                    post_data=post_data,
-                    http_accept=request.axes_http_accept,
-                    path_info=request.axes_path_info,
-                    failures_since_start=1,
-                    attempt_time=request.axes_attempt_time,
-                )
             else:
+                attempt.get_data = Concat("get_data", Value(separator + get_data))
+                attempt.post_data = Concat("post_data", Value(separator + post_data))
+                attempt.http_accept = request.axes_http_accept
+                attempt.path_info = request.axes_path_info
+                attempt.failures_since_start += 1
+                attempt.attempt_time = request.axes_attempt_time
+                attempt.save()
+                
                 log.warning(
-                    "AXES: Username is None and AXES_ONLY_USER_FAILURES is enable, New record won't be created."
+                    "AXES: Repeated login failure by %s. Count = %d of %d. Updating existing record in the database.",
+                    client_str,
+                    attempt.failures_since_start,
+                    get_failure_limit(request, credentials),
                 )
+                
+        else:
+            log.warning(
+                "AXES: Username is None and AXES_ONLY_USER_FAILURES is enable, New record won't be created."
+            )
         if (
             settings.AXES_LOCK_OUT_AT_FAILURE
             and failures_since_start >= get_failure_limit(request, credentials)
```

Comment:
I think we should just use `attempt.failures_since_start` here after adjusting the conditions above a bit?

Reply:
This is still the main check to lock out a user, which did also before not depend on the new entry.